python/caffe/_caffe.cpp

```diff
@@ -248,6 +252,7 @@ BOOST_PYTHON_MODULE(_caffe) {
           bp::return_internal_reference<>()))
     .def("setup", &Layer<Dtype>::LayerSetUp)
     .def("reshape", &Layer<Dtype>::Reshape)
+    .add_property("arguments", &Layer_arguments)
```

Comment:
This is adding an `arguments` property to all `Layer`s and not just `PythonLayer`.